Sources/HTTP/Models/Server/HTTP+Server.swift

```diff
@@ -133,14 +140,15 @@ public final class Server<
                 try serializer.serialize(response)
                 try response.onComplete?(stream)
                 if !request.keepAlive {
+                    self?.streams.remove(stream)
                     try stream.close()
                 }
             } catch ParserError.streamEmpty {
+                self?.streams.remove(stream)
                 try? stream.close()
             } catch {
                 errors(.respond(error))
```

Comment:
I don't think so. The responder or serializer can fail, but that doesn't mean the HTTP connection needs to be closed. This behavior is identical to the blocking version.